components/builder-api/src/server/resources/channels.rs

```diff
@@ -553,6 +559,34 @@ fn get_packages_for_origin_channel_package_version(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn get_packages_for_origin_channel_package_version_head(req: HttpRequest,
+                                                        path: Path<(String,
+                                                              String,
+                                                              String,
+                                                              String)>,
+                                                        pagination: Query<Pagination>)
+                                                        -> HttpResponse {
+    let (origin, channel, pkg, version) = path.into_inner();
+    let channel = ChannelIdent::from(channel);
+
+    let ident = PackageIdent::new(origin, pkg, Some(version), None);
+
+    match do_get_channel_packages(&req, &pagination, &ident, &channel) {
```

Comment:
In my local testing, this would always return 200 OK, even for packages that don't exist.  This is the behavior of the non-head request, so it's probably correct, but it does seem odd to me given the `Err(Error::NotFound)` branch.

Reply:
Huh. I get the same thing. It does seem like here we want to return NOT_FOUND if _count == 0, but that would different from the regular GET. Maybe this should be left alone.

Reply:
I am not seeing that at all:
```
┌─╼[~/development/github/builder] [eeyun/head_routes] 
└────╼ curl --head http://localhost/v1/depot/channels/core/unstable/pkgs/bash/4.4.19/20190115012619
HTTP/1.1 200 OK
Server: nginx/1.17.8
Date: Fri, 13 Mar 2020 15:40:11 GMT
Content-Type: application/json
Content-Length: 0
Connection: keep-alive
cache-control: private, no-cache, no-store
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Alternate-Protocol: 443:npn-spdy/3
X-UA-Compatible: IE=Edge
Strict-Transport-Security: max-age=31536000; includeSubDomains;

┌─╼[~/development/github/builder] [eeyun/head_routes] 
└────╼ curl --head http://localhost/v1/depot/channels/core/unstable/pkgs/bash/4.4.19/20190115012123
HTTP/1.1 404 Not Found
Server: nginx/1.17.8
Date: Fri, 13 Mar 2020 15:40:17 GMT
Content-Length: 0
Connection: keep-alive

Reply:
Ah it looks like a bug with a couple of the routes, will take a closer look.

Reply:
Ok, so the cause of this - is that our `do_get_channel_packages` function returns a tuple of packages and a count of packages. In the case of the standard GET this gets filtered through the `postprocess_channel_package_list` function which constructs a response based on the count. With the GET routes apparently we don't return a 404 if no packages from the request fail which I think is behavior we want to preserve. However the concern for the custom HEAD calls is a bit different. I think in this case we DO want to return a 404 if the search returns nothing. Pushing a commit momentarily to correct this.